app/controllers/CharitableEligibilityController.scala

```diff
@@ -17,32 +17,47 @@
 package controllers
 
 import config.AppConfig
+import controllers.actions.DataRetrievalAction
+import controllers.auth.SessionIdentifierAction
+import controllers.connectors.DataShortCacheConnector
 import forms.EligibilityForm
 import javax.inject.Inject
+import models.YesNoModel
+import play.api.i18n.I18nSupport
 import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
 import uk.gov.hmrc.play.bootstrap.controller.FrontendController
 import views.html.home.eligibility
 
+import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
 
 class CharitableEligibilityController @Inject()(implicit val appConfig: AppConfig,
-                                                mcc: MessagesControllerComponents) extends FrontendController(mcc) {
+                                                sessionIdentifierAction: SessionIdentifierAction,
+                                                dataShortCacheConnector: DataShortCacheConnector,
+                                                getData: DataRetrievalAction,
+                                                mcc: MessagesControllerComponents)
+  extends FrontendController(mcc) with I18nSupport {
 
-  def onPageLoad: Action[AnyContent] = Action.async { implicit request =>
-    Future.successful(Ok(eligibility(EligibilityForm.charitableForm)))
+  def onPageLoad: Action[AnyContent] = (sessionIdentifierAction andThen getData).async { implicit request =>
+
+    val preparedForm = request.userAnswers.flatMap(x => x.eligibilityPurposes).
```

Comment:
Minor thing for future, '(x => x.eligibilityPurposes)' could be '(_.eligibilityPurporses)'